kani-compiler/build.rs

```diff
@@ -62,8 +62,20 @@ pub fn main() {
     // Compile kani library and export KANI_LIB_PATH variable with its relative location.
     let out_dir = env::var("OUT_DIR").unwrap();
     let lib_out = path_str!([&out_dir, "lib"]);
+    let target = env::var("TARGET").unwrap();
     setup_lib(&out_dir, &lib_out, "kani");
     setup_lib(&out_dir, &lib_out, "kani_macros");
     setup_lib(&out_dir, &lib_out, "std");
     println!("cargo:rustc-env=KANI_LIB_PATH={}", lib_out);
+    println!("cargo:rustc-env=TARGET={}", target);
+    println!(
+        "cargo:rustc-env=KANI_EXTERN_OUT_DIR={}",
```

Comment:
Is there a reason why you are not using out_dir?

Reply:
`cargo kani` compiles to the directory with `target/$ARCH/debug/deps` automatically.

I've found the incremental build to be a bit more finichy when using  `target/debug/deps` perhaps because `cargo build --workspace` will also compile to the same directory. I can give that a try.

Reply:
I see. Yeah, that is a problem that we have with our libraries today. We are compiling them twice.

Reply:
Is that one of the issues addressed in your linker update? If so, I can hold off on this one until that is in, and both std/proptest can take advantage.